core/charm/repository/charmhub.go

```diff
@@ -191,6 +196,36 @@ func (c *CharmHubRepository) ResolveWithPreferredChannel(charmURL *charm.URL, re
 	return resCurl, outputOrigin, supportedSeries, nil
 }
 
+// validateFixOrigin, validate the origin and maybe fix as follows:
+//
+//	Platform must have an architecture.
+//	Platform can have empty series AND os.
```

Comment:
I had to think for a couple of seconds here after reading the code below. It might make it easier to read if we make this slightly clearer, like `Platform can have empty series AND empty os`, or `Platform can have both series and os empty ("")`.